deb-get

```diff
@@ -1143,7 +1143,9 @@ if [ "$(id -u)" -ne 0 ]; then
   fancy_message fatal "You must use sudo to run $(basename ${0})."
 fi
 
-if [ "${SUDO_USER}" ]; then
+if [ "$(id -u)" -eq "$(id -ru)" ]; then
```

Comment:
This condition is true for regular users. WHat are you actually trying to test for here?

Reply:
If I am running as root already, and not using `sudo` or `doas` then this block tells me to use `sudo` which I do not need to.

Line 1142 checks if my effective user id is `0`, so I would not have gotten here as a normal user.
This checks if my real user id matches my effective user id.
Maybe `if [ "$(id -ru)" -ne 0 ]; then` here would make it clearer?

Reply:
Why not just comparing with `[ "$(id -u)" -eq 0 ]` ?
```suggestion
if [ "$(id -u)" -eq 0 ]; then
```

Reply:
Because if running with `sudo` or `doas` that also shows true. And that is already checked above.

The change is for the case where the command is actually run as the root user. Using `id -ru` to find out who is behind it.